titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/JobManagerUtil.java

```diff
@@ -137,6 +140,14 @@ public static Task attachNetworkDataFromPod(Task task, PodWrapper podWrapper) {
             contextSetter.accept(TaskAttributes.TASK_ATTRIBUTES_CONTAINER_IPV4, eniIPAddress);
         }
 
+        String hsmAnnotationValue = annotations.get(NETWORK_MODE);
+        if (hsmAnnotationValue.equals("HighScale")) {
+            String subnetAnnotations = annotations.get(NETWORK_SUBNET_IDS);
```

Comment:
Please add one comment as to *why* we  do this, instead of relying on the SG/subnets set in the job